### ISSUE TITLE
[BUG] Cannot create distillery release

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmcloud-otp#388]
Attempting to create a local distillery release now results in:
```
SECRET_KEY_BASE=$(mix phx.gen.secret) MIX_ENV=prod mix distillery.release      
Generated wasmcloud_host app
** (Distillery.Releases.Config.LoadError) could not load release config rel/config.exs
    ** (RuntimeError) could not set configuration via Config. This usually means you are trying to execute a configuration file directly, instead of reading it with Config.Reader
    (elixir 1.13.3) lib/config.ex:91: Config.raise_improper_use!/0
    (elixir 1.13.3) lib/config.ex:190: Config.__env__!/0
    (stdlib 3.17.1) erl_eval.erl:685: :erl_eval.do_apply/6
    (stdlib 3.17.1) erl_eval.erl:893: :erl_eval.expr_list/6
    (stdlib 3.17.1) erl_eval.erl:237: :erl_eval.expr/5
    (stdlib 3.17.1) erl_eval.erl:229: :erl_eval.expr/5
    (stdlib 3.17.1) erl_eval.erl:230: :erl_eval.expr/5
    (stdlib 3.17.1) erl_eval.erl:446: :erl_eval.expr/5
```

This appears to be related to #378 and migrating from `Mix.Config` to `Config`. We may need to upgrade distillery as well

**Comment by brooksmtownsend** [comment]
This appears to be caused by https://github.com/wasmCloud/wasmcloud-otp/blob/main/wasmcloud_host/rel/config.exs#L16, which `config_env()` doesn't work here but can be changed to `:prod` without affecting our workflows

**Comment by brooksmtownsend** [comment]
Fixed by #389
